File: java/Easy/MiddleOfTheLinkedList.java
package Easy;

class ListNode {
        int val;
        ListNode next;
        ListNode() {}
        ListNode(int val) { this.val = val; }
        ListNode(int val, ListNode next) { this.val = val; this.next = next; }
    }

public class MiddleOfTheLinkedList {

    public ListNode middleNode(ListNode head) {
        int n = 0;
        ListNode t = head;
        while(t != null){
            t = t.next;
            n++;
        }
        System.out.println(n);
        n = n/2;
        while(n>0){
            head = head.next;
            n--;
        }
        return head;
    }
}

File: python/Easy/MiddleOFTheLinkedList.py
from typing import Optional

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        n = 0
        t = head
        while t != None:
            t = t.next
            n += 1
        n = n//2
        while n > 0:
            head = head.next
            n -= 1
        return head
    
    def middleNode2(self, head: Optional[ListNode]) -> Optional[ListNode]:
        t = head
        while t and t.next: 
            t = t.next.next
            head = head.next
        return head